sunkit_image/radial.py

```diff
@@ -577,3 +581,101 @@ def fnrgf(
+
+
+def rhef(
+    smap,
+    radial_bin_edges=None,
+    application_radius=0 * u.R_sun,
+    upsilon=0.35,
+    method="inplace",
+):
+    """
+    Implementation of the radial histogram equalizing filter (RHEF).
+
+    The filter works as follows:
+
+    Radial Histogram Equalization is a simple algorithm for removing the radial gradient to reveal
+    coronal structure. It also significantly improves the visualization of high dynamic range solar imagery.
+    It takes the input map and bins the pixels by radius, then ranks each element sequentially and normalizes the pixels at that radius.
+
+    .. note::
+
+        After applying the filter, current plot settings such as the image normalization
+        may have to be changed in order to obtain a good-looking plot.
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        The sunpy map to enhance.
+    radial_bin_edges : `astropy.units.Quantity`
+        A two-dimensional array of bin edges of size ``[2, nbins]`` where ``nbins`` is
+        the number of bins.
+    application_radius : `astropy.units.Quantity`, optional
+        The RHEF is applied to emission at radii above the application_radius.
+        Defaults to 0 solar radii.
+    upsilon : None or int or tuple of size 2, optional
+        A double-sided gamma function applied to the equalized histograms.
+        Equation (TODO) in the paper
+        Defaults to 0.35
+    method: str "inplace", "numpy", "scipy"
+        A string describing which method to use for sorting
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A SunPy map that has had the RHEF applied to it.
+
+    References
+    ----------
+    * Gilly & Cranmer 2024, in prep.
+
+    * The implementation is highly inspired by this doctoral thesis:
+      Gilly, G. Spectroscopic Analysis and Image Processing of the Optically-Thin Solar Corona
+      https://www.proquest.com/docview/2759080511
+    """
+
+    # Get the radii for every pixel
+    map_r = find_pixel_radii(smap).to(u.R_sun)
+
+    # Make sure bins are in the map.
+    if radial_bin_edges is None:
+        radial_bin_edges = equally_spaced_bins(map_r.min(), map_r.max(), map_r.data.shape[0])
+        radial_bin_edges *= u.R_sun
+
+    if radial_bin_edges[1, -1] > np.max(map_r):
+        radial_bin_edges = equally_spaced_bins(
+            inner_value=radial_bin_edges[0, 0],
+            outer_value=np.max(map_r),
+            nbins=radial_bin_edges.shape[1],
+        )
+
+    # Select the sort method
+    if method == "inplace":
+        the_func = percentile_ranks_numpy_inplace
+    elif method == "numpy":
+        the_func = percentile_ranks_numpy
+    elif method == "scipy":
+        the_func = percentile_ranks_scipy
+    else:
+        msg = "{method} is invalid. Allowed values are 'inplace', 'numpy', 'scipy'"
+        raise NotImplementedError(msg)
+
+    # Allocate storage for the filtered data
+    data = np.ones_like(smap.data)
+    meta = smap.meta
+    from tqdm import tqdm
```

Comment:
Can we move this to the top?

Also I need to check if we have a explicit dependency on tqdm.

Reply:
sure, that move is fine. 

and yeah, I was wondering if I should remove the tqdm since the other radial filters don't do that.

Reply:
I think it's worth an issue and maybe rolling it out to any methods with a main for loop.